independent-projects/bootstrap/app-model/src/main/java/io/quarkus/bootstrap/model/PlatformImportsImpl.java

```diff
@@ -92,21 +94,24 @@ public void addPlatformProperties(String groupId, String artifactId, String clas
                         artifactId.length() - BootstrapConstants.PLATFORM_PROPERTIES_ARTIFACT_ID_SUFFIX.length()),
                 version);
         platformImports.computeIfAbsent(bomCoords, c -> new PlatformImport());
-        importedPlatformBoms.computeIfAbsent(groupId, g -> new ArrayList<>()).add(bomCoords);
-
-        final Properties props = new Properties();
-        try (InputStream is = Files.newInputStream(propsPath)) {
-            props.load(is);
-        } catch (IOException e) {
-            throw new AppModelResolverException("Failed to read properties from " + propsPath, e);
-        }
-        for (Map.Entry<?, ?> prop : props.entrySet()) {
-            final String name = String.valueOf(prop.getKey());
-            if (name.startsWith(BootstrapConstants.PLATFORM_PROPERTY_PREFIX)) {
-                if (isPlatformReleaseInfo(name)) {
-                    addPlatformRelease(name, String.valueOf(prop.getValue()));
-                } else {
-                    collectedProps.putIfAbsent(name, String.valueOf(prop.getValue().toString()));
+        importedPlatformBoms.computeIfAbsent(groupId, g -> new ArrayList<>());
+        if (!importedPlatformBoms.get(groupId).contains(bomCoords)) {
+            importedPlatformBoms.get(groupId).add(bomCoords);
```

Comment:
```suggestion
        if (!importedPlatformBoms.get(groupId).add(bomCoords)) {
```

Reply:
If this looks good to you @cdsap , could you please squash the commits? Thanks!

Reply:
Hi @aloubyansky, I was testing your suggestion locally along with the next PR, which adds PlatformImport as an input for the model task. I found that we are again breaking the cacheability of other tasks. The reason is that calling `importedPlatformBoms.get(groupId).add(bomCoords)` does not add a new key to `importedPlatformBoms`; however, it adds a new value entry to the collection for `ArtifactCoords`:
```
private final Map<String, Collection<ArtifactCoords>> importedPlatformBoms = new HashMap<>();
```
As a result, the size of importedPlatformBoms remains the same, but the values for the single entry are increasing. Here is the task input difference in my test.
I can continue with my approach or change importedPlatformBoms to use a `Set` instead of Collection for the `ArtifactCords`

Reply:
Oh, right, sorry, I missed it adding to a list. Ok, please ignore this suggestion.

Reply:
commits squashed